### PR TITLE
Fixed issue #604 - Harshfeudal

### PR DESCRIPTION
**Pull request by harshfeudal**
Done: https://github.com/brainboxdotcc/DPP/issues/604

### CODE REVIEW

File: src/dpp/guild.cpp [hunk 1]
```diff
@@ -111,8 +111,8 @@ std::string guild_member::get_mention() const {
 }
 
 guild_member& guild_member::set_nickname(const std::string& nick) {
-	this->nickname = nick;
-	return *this;
+    this->nickname = nick;
```
Comment (braindigitalis) [inline src/dpp/guild.cpp:114]: please use tabs for indentations

File: src/dpp/guild.cpp [hunk 2]
```diff
@@ -213,13 +213,20 @@ std::string guild_member::build_json(bool with_id) const {
 			j["communication_disabled_until"] = json::value_t::null;
 		}
 	}
-	if (!this->nickname.empty())
+	
+	if (!this->nickname.empty()) {
 		j["nick"] = this->nickname;
+	} else {
+		j["nick"] = find_user(user_id)->username;
```
Comment (Commandserver) [inline src/dpp/guild.cpp:220]: just set the nick to `json::value_t::null` here. no need to search for the user in cache. The method should edit the fields using the passed object.

**Comment by braindigitalis** [comment]
you updated zlib on windows - does this need an update to zlib headers too? what about the .lib file?

**Review by Commandserver — CHANGES_REQUESTED** [review]

**Comment by braindigitalis** [comment]
actually just the .lib is updated not dll or header files so this will cause linking issues...

File: src/dpp/guild.cpp (hunk 2 above)
Comment (harshfeudal) [reply src/dpp/guild.cpp]: ya it still work cool :">

File: include/dpp/user.h [hunk 3]
```diff
@@ -285,11 +285,11 @@ class DPP_EXPORT user : public managed, public json_interface<user>  {
...
 	bool is_verified_bot_dev() const;
 	/**
-	 * @brief Return true if user is a certified moderator
+	 * @brief Return true if user is a Moderator Programs Alumni
 	 *
-	 * @return true if certified moderator
+	 * @return true if Moderator Programs Alumni
 	 */
-	bool is_certified_moderator() const;
+	bool is_moderator_programs() const;
```
[the remark this answers is not in the document]
Comment (harshfeudal) [reply include/dpp/user.h]: Refactored

**Comment by harshfeudal** [comment]
I refactor all, removing old pull request and fix like @Commandserver said :">

**Comment by braindigitalis** [comment]
no mergeable and needs redoing under a new branch

**Pull request closed without merging**